### PR TITLE
use scripted Jenkinsfile to solve version bump issue

### PR DESCRIPTION
**Pull request by pydevops**

### CODE REVIEW

**Comment by pydevops** [comment]
test this please

**Comment by pydevops** [comment]
test this please

**Comment by pydevops** [comment]
test this please

File: Jenkinsfile [hunk 1]
```diff
@@ -33,93 +34,68 @@ metadata:
...
     }
-  }
 
-  post {
-    always {
+  }
+   catch (err) {
+      // if any exception occurs, mark the build as failed
+      currentBuild.result = 'FAILURE'
+      echo "FAILURE"
```
Comment (chrislovecnm) [inline Jenkinsfile]: Do we need this echo?

**Review by chrislovecnm — COMMENTED** [review]
Looks awesome. Just a nit pick.

File: Jenkinsfile (hunk 1 above)
Comment (pydevops) [reply Jenkinsfile]: I think the intent is to say specific about the failure in "Console output" but I am not sure what to say here yet.

Comment (pydevops) [reply Jenkinsfile]: added error output

File: Jenkinsfile [hunk 2]
```diff
@@ -33,93 +34,69 @@ metadata:
...
 
-  post {
-    always {
+  }
+   catch (err) {
+      // if any exception occurs, mark the build as failed
+      // and display a detailed message on the Jenkins console output
+      currentBuild.result = 'FAILURE'
+      echo "FAILURE caught echo ${err}"
+      throw e
```
Comment (chrislovecnm) [inline Jenkinsfile]: Opps missed this ... throw err not e

**Review by chrislovecnm — CHANGES_REQUESTED** [review]

**Comment by pydevops** [comment]
Test this please

**Comment by pydevops** [comment]
Test this please

**Review by chrislovecnm — APPROVED** [review]

**Pull request merged**